src/main/java/org/apache/commons/crypto/stream/CryptoOutputStream.java

```diff
@@ -298,30 +298,39 @@ public boolean isOpen() {
     @Override
     public int write(ByteBuffer src) throws IOException {
         checkStream();
-        final int len = src.remaining();
-        int remaining = len;
+
+        int bytes_written = 0;
+        int remaining = src.remaining();
+
         while (remaining > 0) {
-            final int space = inBuffer.remaining();
-            if (remaining < space) {
+            int position = src.position();
+            int written;
+            final int oldLimit = src.limit();
+            int space = inBuffer.remaining();
+
+            if(remaining < inBuffer.remaining()) {
                 inBuffer.put(src);
-                remaining = 0;
```

Comment:
IMO, after the `src` put into buffer, we should break the loop directly. `encrypt` should only happen when buffer is full or doFinal is called.